### PR TITLE
[bugfix/FARM-508] Fix input alert to start with the current value and auto focus ON

### PR DESCRIPTION
**Pull request by davidfranquet**
[FARM-508]

#### 📲 What

Fix input alert to start with the current value and auto focus ON

#### 🤔 Why
		
Better UX

#### 👀 See

ezgif com-video-to-gif-14

		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/ui/common/InputAlert.dart [hunk 1]
```diff
@@ -216,8 +234,17 @@ class InputAlert extends StatelessWidget {
...
-      _viewModel.confirmInputAction(_textFieldController.text);
+      widget._viewModel.confirmInputAction(_textFieldController.text);
       Navigator.of(context).pop();
     }
   }
+
+  void _selectAllOnFocusListener() {
+    _focusNode.addListener(() {
+      _textFieldController.selection = TextSelection(
+        baseOffset: 0,
```
Comment (alexanishchenko88) [inline lib/ui/common/InputAlert.dart]: Better move it to _Constant I guess

**Review by alexanishchenko88 — COMMENTED** [review]
Not sure if this widget should be statefull or maybe we should create a bloc/provider to this.

**Comment by davidfranquet** [comment]
> Not sure if this widget should be statefull or maybe we should create a bloc/provider to this.

Not necessary. The stateful is only for the textFieldController, because I need to initialise with the incoming value in the initState()

File: lib/ui/common/InputAlert.dart (hunk 1 above)
Comment (davidfranquet) [reply lib/ui/common/InputAlert.dart]: 👍

Comment (davidfranquet) [reply lib/ui/common/InputAlert.dart]: DOne

**Comment by alexanishchenko88** [comment]
@DavidFD005 in this case we can leave it as is

**Review by alexanishchenko88 — APPROVED** [review]

**Review by albertmiro — APPROVED** [review]

**Pull request merged**